rskj-core/src/main/java/co/rsk/mine/gas/provider/MinGasPriceProviderFactory.java

```diff
@@ -17,25 +16,26 @@ private MinGasPriceProviderFactory() {
 
     public static MinGasPriceProvider create(long fixedMinGasPrice, StableMinGasPriceSystemConfig stableMinGasPriceSystemConfig, OnChainMinGasPriceProvider.GetContextCallback getContextCallback) {
         FixedMinGasPriceProvider fixedMinGasPriceProvider = new FixedMinGasPriceProvider(fixedMinGasPrice);
+
         if (stableMinGasPriceSystemConfig == null) {
             logger.warn("Could not find stable min gas price system config, using {} provider", fixedMinGasPriceProvider.getType().name());
             return fixedMinGasPriceProvider;
         }
         if (!stableMinGasPriceSystemConfig.isEnabled()) {
             return fixedMinGasPriceProvider;
         }
+
         MinGasPriceProviderType method = stableMinGasPriceSystemConfig.getMethod();
         if (method == null) {
             logger.error("Could not find valid method in config, using fallback provider: {}", fixedMinGasPriceProvider.getType().name());
             return fixedMinGasPriceProvider;
         }
 
-
         switch (method) {
             case WEB:
-                return WebStableMinGasPriceProviderFactory.create(stableMinGasPriceSystemConfig, fixedMinGasPriceProvider);
+                return new WebMinGasPriceProvider(stableMinGasPriceSystemConfig, fixedMinGasPriceProvider);
             case ON_CHAIN:
-                return OnChainMinGasPriceProviderFactory.create(stableMinGasPriceSystemConfig, fixedMinGasPriceProvider, getContextCallback);
+                return new OnChainMinGasPriceProvider(fixedMinGasPriceProvider, stableMinGasPriceSystemConfig.getMinStableGasPrice(), stableMinGasPriceSystemConfig.getOnChainConfig(), getContextCallback);
```

Comment:
```suggestion
                return new OnChainMinGasPriceProvider(stableMinGasPriceSystemConfig, fixedMinGasPriceProvider, getContextCallback);
```

WDYT on updating this constructor to receive the whole config instead of the separated values? So it uses the same strategy as the `WebMinGasPriceProvider` constructor? Or we could explode the config on the `WebMinGasPriceProvider` too so it's similar to the `OnChainMinGasPriceProvider` right now?